src/runner/browser-env/vite/browser-modules/driver.ts

```diff
@@ -0,0 +1,173 @@
+import {
+    WebDriverProtocol,
+    MJsonWProtocol,
+    JsonWProtocol,
+    AppiumProtocol,
+    ChromiumProtocol,
+    SauceLabsProtocol,
+    SeleniumProtocol,
+    GeckoProtocol,
+    WebDriverBidiProtocol,
+} from "@wdio/protocols";
+import { webdriverMonad, sessionEnvironmentDetector, devtoolsEnvironmentDetector } from "@wdio/utils";
+import { getEnvironmentVars } from "webdriver";
+import { SOCKET_MAX_TIMEOUT, SOCKET_TIMED_OUT_ERROR, MAX_ARGS_LENGTH } from "./constants.js";
+import { BrowserEventNames } from "./types.js";
+
+type ProtocolCommandFn = (...args: unknown[]) => Promise<unknown>;
+type PropertiesObject = Record<string, PropertyDescriptor>;
+
+const COMMANDS_HANDLED_BY_SERVER = ["debug", "saveScreenshot", "savePDF"];
+
+export default class ProxyDriver {
+    static newSession(
+        params: Record<string, unknown>,
+        modifier: never,
+        userPrototype: Record<string, PropertyDescriptor>,
+        commandWrapper: VoidFunction | undefined,
+    ): unknown {
+        const monad = webdriverMonad(params, modifier, getWdioPrototype(userPrototype));
+        return monad(window.__testplane__.sessionId, commandWrapper);
+    }
+}
+
+function getWdioPrototype(userPrototype: Record<string, PropertyDescriptor>): PropertiesObject {
+    for (const commandName of COMMANDS_HANDLED_BY_SERVER) {
+        delete userPrototype[commandName];
```

Comment:
delete commands implementation because they must be mocked (inside `getMockedProtocolCommands`)